src/transaction-flow/input/ExtendNames/ExtendNames-flow.tsx

```diff
@@ -44,81 +79,232 @@ const OptionBar = styled.div(
+
+  if (expiryLoading) return null
+  return (
+    <NamesListItemContainer>
+      <NamesListItemAvatarWrapper>
+        <Avatar src={avatar || zorb} label={name} />
+      </NamesListItemAvatarWrapper>
+      <NamesListItemContent>
+        <NamesListItemTitle>
+          <StyledName name={name} />
+        </NamesListItemTitle>
+        {expiry?.expiry && (
+          <NamesListItemSubtitle>
+            <ShortExpiry expiry={expiry.expiry} textOnly />
+          </NamesListItemSubtitle>
+        )}
+      </NamesListItemContent>
+    </NamesListItemContainer>
+  )
+}
+
+const NamesListContainer = styled.div(
+  ({ theme }) => css`
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+    gap: ${theme.space['2']};
+  `,
+)
+
+type NamesListProps = {
+  names: string[]
+}
+
+const NamesList = ({ names }: NamesListProps) => {
+  return (
+    <NamesListContainer data-testid="extend-names-names-list">
+      {names.map((name) => (
+        <NamesListItem key={name} name={name} />
+      ))}
+    </NamesListContainer>
+  )
+}
+
 type Data = {
   names: string[]
+  isSelf?: boolean
 }
 
 export type Props = {
   data: Data
 } & TransactionDialogPassthrough
 
-const ExtendNames = ({ data: { names }, dispatch, onDismiss }: Props) => {
+const ExtendNames = ({ data: { names, isSelf }, dispatch, onDismiss }: Props) => {
   const { t } = useTranslation('transactionFlow')
 
-  const fiatUnit = 'usd'
+  const [view, setView] = useState<'name-list' | 'registration'>(
+    names.length > 1 ? 'name-list' : 'registration',
+  )
 
   const [years, setYears] = useState(1)
+  const duration = yearsToSeconds(years)
+
   const [currencyUnit, setCurrencyUnit] = useState<CurrencyUnit>('eth')
+  const fiatUnit = 'usd'
   const currencyDisplay = currencyUnit === 'fiat' ? fiatUnit : 'eth'
 
-  const { data: transactionData, isLoading: transactionDataLoading } =
-    useEstimateTransactionCost('RENEW')
-  const { gasPrice, transactionFee } = transactionData || {}
-  const gasLabel = gasPrice ? `${formatUnits(gasPrice, 'gwei')} gwei` : '-'
-
-  const { data: ethPrice, loading: ethPriceLoading } = useEthPrice()
+  const { base: rentFee, loading: priceLoading } = usePrice(names, true)
 
-  const rentFee = ethPrice
-    ? BigNumber.from('5000000000000000000').div(ethPrice).div(1e8)
-    : undefined
   const totalRentFee = rentFee ? rentFee.mul(years) : undefined
+  const transactions = [
+    makeTransactionItem('extendNames', { names, duration, rentPrice: totalRentFee!, isSelf }),
+  ]
+
+  const {
+    gasLimit: estimatedGasLimit,
+    error: estimateGasLimitError,
+    isLoading: isEstimateGasLoading,
+  } = useEstimateGasLimitForTransactions(transactions)
+  console.log(estimateGasLimitError)
+
+  const hardcodedGasLimit = gasLimitDictionary.RENEW(names.length)
+  const gasLimit = estimatedGasLimit || hardcodedGasLimit
+
+  const { data: feeData, isLoading: isFeeDataLoading } = useFeeData()
+  const gasPrice = feeData?.maxFeePerGas || undefined
```

Comment:
Doesn't the optional chaining operator return undefined if it's falsey anyway?

Reply:
You know I'm American so I need things to be very specific so I'll understand

Reply:
Its a typescript thing. feeData from 'wagmi' can also return null.